### ISSUE TITLE
gather() method runs two times then stops with no error

### ISSUE THREAD

**Issue by keenjo** [huggingface/accelerate#566]
I've implemented the gather() method in the validation loop of my script but it seems as if the program will run through the loop two times and then stop but the program doesn't fail and thus no error appears. Here is my validation loop below:

```
        for itr, data in tqdm(enumerate(valid_dataloader), desc='Validation loop'):

             src_input_ids, src_attn_mask, tgt_input_ids, tgt_attn_mask = data
             
             labels = shift_target_inputs_to_labels(tgt_input_ids, valid_dataset.tokenizer.pad_token_id)
             with torch.no_grad():
                 output = model(
                     input_ids=src_input_ids,
                     attention_mask=src_attn_mask,
                     decoder_input_ids=tgt_input_ids,
                     decoder_attention_mask=tgt_attn_mask,
                     labels=labels
                    )
             
             output = accelerator.gather(output) # Gathering all of the outputs with huggingface accelerator
             valid_loss = output[0]
             valid_epoch_sum_loss += valid_loss * src_input_ids.shape[0]
```

**Comment by keenjo** [comment]
It seems as if I didn't need to use `gather` here because I wasn't computing any evaluation metrics. I thought I would need to combine the validation loss at each epoch but it turns out it wasn't necessary.
